Morph2/src/core/net/sf/morph2/Defaults.java

```diff
@@ -42,7 +43,10 @@
  * @since Jan 9, 2005
  */
 public abstract class Defaults {
-
+	
+	// making these singletons to save memory
+	public static final TextToNumberConverter TEXT_TO_NUMBER_CONVERTER = new TextToNumberConverter();
```

Comment:
How confident are you that this is thread-safe? I poked around it a little, and `BaseTransformer` has some instance variables that may or may not be thread-safe. They might be, but I didn't want to keep digging if you'd looked at it already.